commit 7825364d4d6bfbe6c27af3a6ea38b36d20738e28 (refs/changes/37/2837/2)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2017-10-25T01:48:08+00:00 (1 year, 11 months ago)
    
    ADMINUI-2370 Disk provisioned is misleading as it includes the reserved amount

diff --git a/less/server.less b/less/server.less
index 97057294..8c962e5a 100644
--- a/less/server.less
+++ b/less/server.less
@@ -22,6 +22,7 @@
   .provisionable-disk,
   .provisioned-disk,
   .total-disk,
+  .unreserved-disk,
   .reserved-disk {
     .make-sm-column(2, 0);
 
@@ -31,7 +32,7 @@
     .title { text-transform: uppercase; color: #666;  font-size: 12px; }
   }
 
-  .total-disk { opacity: 0.5; }
+  .unreserved-disk, .total-disk { opacity: 0.5; }
 
   .memory-overview h5 {
     text-transform: uppercase;
@@ -67,6 +68,7 @@
 
   .provisionable-disk .value,
   .provisioned-disk .value,
+  .reserved-disk .value,
   .provisionable-memory .value,
   .provisioned-memory .value,
   .reserved-memory .value {
@@ -77,14 +79,18 @@
   .provisioned-disk .value,
   .provisionable-memory .value,
   .provisioned-memory .value,
+  .reserved-disk .value,
   .reserved-memory .value,
   .total-disk .value,
+  .unreserved-disk ,value,
   .unreserved-memory .value,
   .total-memory .value {
     color: #888;
   }
 
   .unreserved-memory, .total-memory { opacity: 0.5; }
+
+  .unreserved-disk,
   .unreserved-memory {
     border-left: 1px solid #aaa;
   }
diff --git a/www/js/components/pages/server/disk-overview.jsx b/www/js/components/pages/server/disk-overview.jsx
index 8acd8ac5..b2150f5a 100644
--- a/www/js/components/pages/server/disk-overview.jsx
+++ b/www/js/components/pages/server/disk-overview.jsx
@@ -12,6 +12,7 @@ var React = require('react');
 var BackboneMixin = require('../../_backbone-mixin');
 var ServerDiskUtilizationCircle = require('./disk-utilization-circle');
 var utils = require('../../../lib/utils');
+var POOL_USABLE_RATIO = 0.94;
 
 var ServerDiskOverview = React.createClass({
     mixins: [BackboneMixin],
@@ -24,7 +25,12 @@ var ServerDiskOverview = React.createClass({
             return null;
         }
 
-        var provisioned = server.disk_pool_size_bytes - (server.unreserved_disk * 1048576);
+//      DAPI applies a pool usage ratio in its calculation of provisionable disk.
+//      Unfortunately that's named unreserved_disk which is different from what it means
+//      in this context. Here "unreserved" means what is not reserved by the system.
+        var reserved = server.disk_pool_size_bytes * (1 - POOL_USABLE_RATIO);
+        var unreserved = server.disk_pool_size_bytes * POOL_USABLE_RATIO;
+        var provisioned = unreserved - (server.unreserved_disk * 1048576);
         var provisionable = server.unreserved_disk * 1048576;
         var total = server.disk_pool_size_bytes;
         
@@ -37,6 +43,8 @@ var ServerDiskOverview = React.createClass({
 
         var provisioned = utils.getReadableSize(provisioned);
         var provisionable = utils.getReadableSize(provisionable);
+        var reserved = utils.getReadableSize(reserved);
+        var unreserved = utils.getReadableSize(unreserved);
         var total = utils.getReadableSize(server.disk_pool_size_bytes);
 
         return <div className="disk-overview">
@@ -57,7 +65,14 @@ var ServerDiskOverview = React.createClass({
                     <div className="value">{provisioned.value + ' ' + provisioned.measure}</div>
                     <div className="title">Provisioned</div>
                 </div>
-
+                <div className="reserved-disk">
+                    <div className="value">{reserved.value + ' ' + reserved.measure}</div>
+                    <div className="title">Reserved</div>
+                </div>
+                <div className="unreserved-disk">
+                    <div className="value">{unreserved.value + ' ' + unreserved.measure}</div>
+                    <div className="title">Unreserved</div>
+                </div>
                 <div className="total-disk">
                     <div className="value">{total.value + ' ' + total.measure}</div>
                     <div className="title">Total</div>
